indico/modules/events/editing/client/js/selectors.js

```diff
@@ -118,3 +119,22 @@ export function processRevisions(revisions) {
 
   return newRevisions;
 }
+
+export const getDetails = state => state.timeline.details;
+export const isLoading = state => state.timeline.isLoading;
+export const getTimelineBlocks = state => state.timeline.timelineBlocks;
+export const getLastTimelineBlock = createSelector(
+  getTimelineBlocks,
+  blocks => blocks && blocks[blocks.length - 1]
+);
+export const getLastRevision = createSelector(
+  getDetails,
+  details => details && details.revisions[details.revisions.length - 1]
+);
+export const getLastState = createSelector(
+  getLastRevision,
+  lastRevision =>
+    lastRevision &&
+    (lastRevision.finalState.name === 'none' ? lastRevision.initialState : lastRevision.finalState)
```

Comment:
Let's use the enum here (ie `FinalRevisionState.none`)?

Reply:
There is already an enum for that ;)